lib/src/main/java/com/futurewei/alcor/common/db/ignite/IgniteDbCache.java

```diff
@@ -19,63 +19,64 @@
 import com.futurewei.alcor.common.db.CacheException;
 import com.futurewei.alcor.common.db.ICache;
 import com.futurewei.alcor.common.db.Transaction;
+import com.futurewei.alcor.common.db.query.CachePredicate;
+import com.futurewei.alcor.common.db.query.ScanQueryBuilder;
+import com.futurewei.alcor.common.db.query.impl.MapPredicate;
 import com.futurewei.alcor.common.logging.Logger;
 import com.futurewei.alcor.common.logging.LoggerFactory;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.binary.BinaryObject;
 import org.apache.ignite.cache.query.Query;
 import org.apache.ignite.cache.query.QueryCursor;
 import org.apache.ignite.cache.query.ScanQuery;
-import org.apache.ignite.client.ClientCache;
-import org.apache.ignite.client.ClientCacheConfiguration;
 import org.apache.ignite.client.ClientException;
-import org.apache.ignite.client.IgniteClient;
 import org.springframework.util.Assert;
 
 import javax.cache.Cache;
 import javax.cache.expiry.ExpiryPolicy;
+import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.logging.Level;
 import java.util.stream.Collectors;
 
 
-public class IgniteCache<K, V> implements ICache<K, V> {
+public class IgniteDbCache<K, V> implements ICache<K, V> {
     private static final Logger logger = LoggerFactory.getLogger();
-    private ClientCache<K, V> cache;
-    private IgniteClient igniteClient;
+
+    private static final int RESULT_THRESHOLD_SIZE = 100000;
+    private IgniteCache<K, V> cache;
     private IgniteTransaction transaction;
 
-    public IgniteCache(IgniteClient igniteClient, String name) {
-        this.igniteClient = igniteClient;
+    public IgniteDbCache(Ignite ignite, String name) {
 
         try {
-            cache = igniteClient.getOrCreateCache(name);
+            cache = ignite.getOrCreateCache(name);
         } catch (ClientException e) {
```

Comment:
Since we have used Ignite instead of IgniteClient, the exceptions of cache operations should be modified accordingly.